### PR TITLE
added precision handling for non-integers, added geometry handler

### PR DESCRIPTION
**Pull request by Danlock**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,10 +11,16 @@ sequelizeToJoi(model, [options = {}])
 ### Example:
 
 ```
-import sequelizeToJoi from '@revolttv/sequelize-to-joi';
+const sequelizeToJoi = require('sequelize-to-joi');
```
Comment (andrewburgess) [inline README.md]: Leave as `import`. Also, our package is published under our org's NPM account, so the `@revolttv/` part is needed

File: src/map.js [hunk 2]
```diff
@@ -1,5 +1,44 @@
-import _   from 'lodash';
-import Joi from 'joi';
+'use strict';
```
Comment (andrewburgess) [inline src/map.js]: Don't need `use strict` for ES6 modules

File: src/map.js [hunk 3]
```diff
@@ -1,5 +1,44 @@
...
+const _   = require('lodash');
+const Joi = require('joi');
```
Comment (andrewburgess) [inline src/map.js]: Would prefer keeping the `import` statements

File: src/map.js [hunk 4]
```diff
@@ -1,5 +1,44 @@
...
+
+const VALID_GEOJSON_TYPES = ['Point', 'Linestring', 'Polygon','MultiPoint','MultiLineString','MultiPolygon','GeometryCollection','FeatureCollection','Feature'];
```
Comment (andrewburgess) [inline src/map.js]: Probably best to split these out onto new lines

File: src/map.js [hunk 5]
```diff
@@ -1,5 +1,44 @@
...
+function createGeoJSONValidator() {
+    let geojsonBasic = Joi.object({
+        type: Joi.string().valid(VALID_GEOJSON_TYPES).insensitive(),
+        coordinates: Joi.array().sparse(),
+        bbox: Joi.array().sparse(),
+        properties: Joi.object(),
+        crs: Joi.object({
+            type: Joi.string().valid('name','link').insensitive().required(),
+            properties: Joi.object().required(),
+        }),
```
Comment (andrewburgess) [inline src/map.js]: No trailing commas

File: src/map.js [hunk 6]
```diff
@@ -1,5 +1,44 @@
...
+        geometries: Joi.array().items(geojsonBasic).sparse(),
+        features: Joi.array().items(geojsonBasic).sparse(),
+    });
+}
+
+function createNonIntegerValidator(attr) {
+    let j = Joi.number();
+    if (!attr.type) return j;
+    if (attr.type._length) {
+        let bound = Math.pow(10,attr.type._length);
```
Comment (andrewburgess) [inline src/map.js]: Space after comma

File: src/map.js [hunk 7]
```diff
@@ -1,5 +1,44 @@
...
+    return geojsonBasic.keys({
+        geometry: geojsonBasic,
+        geometries: Joi.array().items(geojsonBasic).sparse(),
+        features: Joi.array().items(geojsonBasic).sparse(),
+    });
+}
+
+function createNonIntegerValidator(attr) {
+    let j = Joi.number();
+    if (!attr.type) return j;
```
Comment (andrewburgess) [inline src/map.js]: For consistency, wrap in `{` `}` and put `return` on newline

File: src/map.js [hunk 8]
```diff
@@ -1,5 +1,44 @@
...
+    if (attr.type._length) {
+        let bound = Math.pow(10,attr.type._length);
+        j = j.less(bound).greater(-1 * bound);
+    }
+    if (attr.type._decimals) j = j.precision(attr.type._decimals).strict();
```
Comment (andrewburgess) [inline src/map.js]: Same as above, wrap in `{` `}` and put inner block on new line

File: src/map.js [hunk 9]
```diff
@@ -70,7 +110,12 @@ function mapValidator(joi, validator, key) {
     }
 }
 
-export default function (attribute) {
+module.exports = function (attribute) {
+    //allow user to personally set joi objects in models, mainly for JSON/B data types
+    if (attribute.sequelize_to_joi_override) {
```
Comment (andrewburgess) [inline src/map.js]: Would prefer this to be `camelCase`

File: src/map.js [hunk 10]
```diff
@@ -70,7 +110,12 @@ function mapValidator(joi, validator, key) {
     }
 }
 
-export default function (attribute) {
+module.exports = function (attribute) {
```
Comment (andrewburgess) [inline src/map.js]: Leave as `export default`

File: test/index.js [hunk 11]
```diff
@@ -1,3 +1,4 @@
+'use strict';
```
Comment (andrewburgess) [inline test/index.js]: Don't need `use strict`

File: test/index.js [hunk 12]
```diff
@@ -118,4 +169,93 @@ describe('sequelize-to-joi', () => {
...
+    it('should pass proper geojson object', () => {
+        let schema = sequelizeToJoi(sequelize.models.Geo, { omitAutoGenerated: true });
+        let test = {
+            geojson : {
+                type: "FeatureCollection",
+                features: [{ 
+                    "type": "Feature",
+                    "geometry": {"type": "Point", "coordinates": [102.0, 0.5]},
+                    "properties": {"prop0": "value0"}
+                },{
```
Comment (andrewburgess) [inline test/index.js]: Space after comma

File: test/index.js [hunk 13]
```diff
@@ -118,4 +169,93 @@ describe('sequelize-to-joi', () => {
...
+            geojson : {
+                type: "FeatureCollection",
+                features: [{ 
+                    "type": "Feature",
+                    "geometry": {"type": "Point", "coordinates": [102.0, 0.5]},
+                    "properties": {"prop0": "value0"}
+                },{ 
+                    "type": "Feature",
+                    "geometry": {
+                      "type": "LineString",
```
Comment (andrewburgess) [inline test/index.js]: Indentation is a bit off in here. Make sure that `npm run lint` is passing

File: README.md (hunk 1 above)
Comment (Danlock) [inline README.md]: whoops, i made these changes on my personal branch and ported them over, forgot about these. stylistic differences. Will change now.

File: test/index.js [hunk 14]
```diff
@@ -118,4 +168,92 @@ describe('sequelize-to-joi', () => {
...
+                    "geometry": {
+                        "type": "LineString",
+                        "coordinates": [100.0, 1.0],
+
+                    },
+                    "properties": {
+                        "prop0": "value0",
+                        "prop1": 0.0
+                    }
+                    }, {
```
Comment (andrewburgess) [inline test/index.js]: Getting some linter errors about the indentation for these geojson definitions

File: test/index.js [hunk 15]
```diff
@@ -118,4 +168,92 @@ describe('sequelize-to-joi', () => {
...
+                        "type": "LineString",
+                        //coordinates should be an array
+                        "coordinates": 1,
+
+                    },
+                    "properties": {
+                        "prop0": "value0",
+                        "prop1": 0.0
+                    }
+                    }, {
```
Comment (andrewburgess) [inline test/index.js]: Getting some linter errors about the indentation for these geojson definitions

**Pull request merged**